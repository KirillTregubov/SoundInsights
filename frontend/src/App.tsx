import {
  QueryClientProvider,
  QueryClient,
  useQuery
} from '@tanstack/react-query'

import { getQuery } from 'lib/api'

const App: React.FC = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false
      }
    }
  })

  return (
    <QueryClientProvider client={queryClient}>
      <div id="App">
        <div className="m-2">Hello from App.tsx</div>
        <Fetch />
      </div>
    </QueryClientProvider>
  )
}

const Fetch: React.FC = () => {
  const query = useQuery(['query'], getQuery)

  return (
    <div className="m-2 rounded-lg bg-gray-200 p-4">
      <h1 className="font-semibold ">Fetch data from backend</h1>
      <div>Data: {JSON.stringify(query.data)}</div>
      <h4 className="mt-1 text-sm text-gray-900">
        Received at {new Date(query.dataUpdatedAt).toLocaleString()}
      </h4>
    </div>
  )
}

export default App
