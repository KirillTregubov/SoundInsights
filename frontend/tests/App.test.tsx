import { render } from '@testing-library/react'

import App from 'src/App'

describe('test App.tsx', () => {
  test('contains header', () => {
    const { getByText } = render(<App />)
    const header = getByText(/Team Ez2Type/i)
    expect(header).toBeInTheDocument()
  })

  test('contains RecommendedTracks component', () => {
    const { getByText } = render(<App />)
    const header = getByText(/Get Music Recommendations/i)
    expect(header).toBeInTheDocument()
  })
})
