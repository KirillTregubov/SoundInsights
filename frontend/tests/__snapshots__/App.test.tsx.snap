// Vitest Snapshot v1

exports[`App.tsx > renders correctly 1`] = `
<div
  id="App"
>
  <div
    className="m-2"
  >
    Hello from App.tsx
  </div>
  <div
    className="m-2 rounded-lg bg-gray-200 p-4"
  >
    <h1
      className="font-semibold "
    >
      Fetch data from backend
    </h1>
    <div>
      Data: 
    </div>
    <h4
      className="mt-1 text-sm text-gray-900"
    >
      Received at 
      1969-12-31, 7:00:00 p.m.
    </h4>
  </div>
</div>
`;
